code/game/objects/items/tcg/tcg.dm

```diff
@@ -0,0 +1,273 @@
+///A global list of cards, or rather changes to be applied to cards in the format
+
+GLOBAL_LIST_EMPTY_TYPED(card_list, /datum/card)
+
+/obj/item/tcgcard
+	name = "Coder"
+	desc = "Wow, a mint condition coder card! Better tell the Github all about this!"
+	icon = 'icons/obj/tcg.dmi'
+	icon_state = "runtime"
+	w_class = WEIGHT_CLASS_TINY
+	var/id = -1 //Unique ID, for use in lookups and storage, used to index the global datum list where the rest of the card's info is stored
+	var/flipped = 0
+
+/obj/item/tcgcard/Initialize(mapload, datum/card/temp)
+	. = ..()
+	name = temp.name
+	desc = temp.desc
+	icon = icon(temp.icon)
+	icon_state = temp.icon_state
+	id = temp.id
+	transform = matrix(0.3,0,0,0,0.3,0)
+
+/datum/card
+	var/id = -1 //Unique ID, for use in lookups and (eventually) for persistence. MAKE SURE THIS IS UNIQUE FOR EACH CARD, OR THE ENTIRE SYSTEM WILL BREAK, AND I WILL BE VERY DISAPPOINTED.
```

Comment:
Could you use the card name or type path as the ID? Relying on the coders to give it a unique number and failure of that causing issues is dangerous.

Reply:
We'd have the same problem then right? Someone could add a duplicate name card.

I agree that it will be a bit of a problem if this gets expanded on in future, but I think it might be better to reset the ids when changing series.

Reply:
It's also worth mentioning that the desc there is a little overblown. By the nature of how we are loading this, id conflicts will at most cause a rarity double entry and an override of the old card.

Reply:
Not if you add it on initialize, automatically, for example.
Or just used it on the code elsewhere by the type path instead.

Reply:
I've changed it to be a global list of series, with separate lists for each rarity in the series containing card indexes, matched against a list of all the cards in a series.

Reply:
I don't think we could use typepath, because the definitions of the cards get loaded in from the json files. I've done my best to make maintaining those noncomplex.

Reply:
👍

Reply:
Actually you are stringifying `id` and using it as an assoc-list index. Why don't you just use the card's name as defined by the json, which are supposedly unique? What do you need a stringified number for?

Reply:
I don't trust coders to make unique names, in fact if memory serves we have at least one duplicate currently

Reply:
Also I'd like to allow for duplicate names, it seems like a nice thing to have

Reply:
Then maybe do similar to signals, in where we have unique string identifiers which are clear references to the thing they represent, instead of magic numbers?

Reply:
Is your concern understand-ability? It's semi simple at this point, the number goes up card by card. I guess I don't really understand the issue with using uids. If someone has to manually write these out I'd like to make it as easy as possible.

I guess we could automate id assigning, but I'd like to allow for ids to be out of order, in case we need to depreciate a card or something.

Reply:
Mostly it's an issue with brittle code. If you have a series of magic numbers the coders need to input correctly, failure on doing so creating bugs, then something is up.

What do you need the numbers for? Do they help with a certain card order? If so you could add an order variable and sort the list using them, which wouldn't cause serious issues on repeated values.